sympy/combinatorics/rewritingsystem.py

```diff
@@ -369,17 +369,22 @@ def _add_to_automaton(self, rules):
             len_letter_word_array = len(letter_word_array)
             for i in range (1, len_letter_word_array):
                 letter_word_array[i] = letter_word_array[i-1]*letter_word_array[i]
+                # Add accept states.
+                elem = letter_word_array[i-1]
+                if not elem in self.reduction_automaton.states:
+                    self.reduction_automaton.add_state(elem, state_type='a')
+                    accept_states.append(elem)
             proper_prefixes[rule] = letter_word_array
+            # Check for overlaps between dead and accept states.
+            if rule in accept_states:
+                self.reduction_automaton.states[rule].state_type = 'd'
+                self.reduction_automaton.states[rule].rh_rule = all_rules[rule]
+                accept_states.remove(rule)
```

Comment:
Not quite what I had in mind but I suppose this works too - the error is gone and hopefully this is slightly more efficient.